website/src/roadmap/portability.md

```diff
@@ -170,18 +170,32 @@ Python streaming mode is not yet supported on Spark.
 The Beam Python SDK allows configuration of the SDK harness to accommodate varying cluster setups.
 
 - `environment_type` determines where user code will be executed.
-  - `LOOPBACK`: User code is executed within the same process that submitted the pipeline. This
-    option is useful for local testing. However, it is not suitable for a production environment,
-    as it requires a connection between the original Python process and the worker nodes, and
-    performs work on the machine the job originated from, not the worker nodes.
-  - `PROCESS`: User code is executed by processes that are automatically started by the runner on
-    each worker node.
+  `environment_config` configures the environment depending on the value of `environment_type`.
   - `DOCKER` (default): User code is executed within a container started on each worker node.
-    This requires docker to be installed on worker nodes. For more information, see
+    This requires docker to be installed on worker nodes.
+    - `environment_config`: URL for the Docker container image. Official Docker images
+    are available [here](https://hub.docker.com/u/apachebeam) and are used by default.
+    Alternatively, you can build your own image by following the instructions
     [here]({{ site.baseurl }}/documentation/runtime/environments/).
-- `environment_config` configures the environment depending on the value of `environment_type`.
-  - When `environment_type=DOCKER`: URL for the Docker container image.
-  - When `environment_type=PROCESS`: JSON of the form `{"os": "<OS>", "arch": "<ARCHITECTURE>",
+  - `PROCESS`: User code is executed by processes that are automatically started by the runner on
+    each worker node.
+    - `environment_config`: JSON of the form `{"os": "<OS>", "arch": "<ARCHITECTURE>",
     "command": "<process to execute>", "env":{"<Environment variables 1>": "<ENV_VAL>"} }`. All
     fields in the JSON are optional except `command`.
+      - For `command`, it is recommended to use the bootloader executable, which can be built from
+        source with `./gradlew :sdks:python:container:build` and copied from
+        `sdks/python/container/build/target/launcher/linux_amd64/boot` to worker machines.
+        Note that the Python bootloader assumes Python and the `apache_beam` module are installed
+        on each worker machine.
+  - `EXTERNAL`: User code will be dispatched to an external service. For example, one can start
+    an external service for Python workers by running
+    `docker run -p=50000:50000 apachebeam/python3.6_sdk --worker_pool`.
+    - `environment_config`: Address for the external service, e.g. `localhost:50000`.
+    - To access a Dockerized worker pool on Mac or Windows, set the `BEAM_WORKER_POOL_IN_DOCKER_VM`
+      environment variable: `export BEAM_WORKER_POOL_IN_DOCKER_VM=1`.
```

Comment:
Doesn't this have to be set for the container? Not sure that is clear here.

Reply:
This needs to be set for the client of the containerized worker pool, i.e. job server, Flink task manager etc.

Reply:
Clarified this point.